components/backup/src/endpoint.rs

```diff
@@ -1961,6 +1960,58 @@ pub mod tests {
         }
     }
 
+    #[test]
+    fn test_seek_ranges_2() {
+        let (_tmp, endpoint) = new_endpoint();
+
+        endpoint.region_info.set_regions(vec![
+            (b"2".to_vec(), b"4".to_vec(), 1),
+            (b"6".to_vec(), b"8".to_vec(), 2),
+        ]);
+        let sub_ranges: Vec<(&[u8], &[u8])> = vec![(b"1", b"11"), (b"3", b"7"), (b"8", b"9")];
+        let expect: Vec<(&[u8], &[u8])> = vec![(b"3", b"4"), (b"6", b"7")];
+
+        let mut ranges = Vec::with_capacity(sub_ranges.len());
+        for &(start_key, end_key) in &sub_ranges {
+            let start_key = (!start_key.is_empty()).then_some(Key::from_raw(start_key));
+            let end_key = (!end_key.is_empty()).then_some(Key::from_raw(end_key));
+            ranges.push((start_key, end_key));
+        }
+        let mut prs = Progress::new_with_ranges(
+            endpoint.store_id,
+            ranges,
+            endpoint.region_info.clone(),
+            KeyValueCodec::new(false, ApiVersion::V1, ApiVersion::V1),
+            engine_traits::CF_DEFAULT,
+        );
+
+        let mut ranges = Vec::with_capacity(expect.len());
+        loop {
+            let n = (rand::random::<usize>() % 2) + 1;
+            let mut r = match prs.forward(n, false) {
+                None => break,
+                Some(r) => r,
```

Comment:
Should explicitly cover the Some(empty Vec) case.

Reply:
done